### PR TITLE
Show deleted clusters with the 'gsctl list clusters' command

### PR DESCRIPTION
**Pull request by marians**
Towards https://github.com/giantswarm/giantswarm/issues/7516

The goal of that story is to raise awareness for clusters still using resources after deletion has been initialised, until all resources are actually removed.

This adds a `--show-deleting` flag to the command and a conditional `DELETING SINCE` column to the cluster listing, showing the deletion date if available. Deleted clusters are also marked red when deleted more than 24 hours ago.

Note: When using `-o json` for JSON output, deleted clusters will be listed. The `--show-deleted` flag is not applicable in that case.

### Preview

image

image

### CODE REVIEW

**Comment by xh3b4sd** [comment]
Just some thoughts to consider. 

* `docker ps -a` shows dead containers. Do we want to show deleted clusters using a flag or always? 
* Initially I thought there should maybe two tables be shown to have a more clear visual separation. Uncertain if this is a good practise. 
* Should there be some kind of sorting? Should deleted clusters be shown first, or last? The list of Cloud Formations in the AWS Console does not change sorting because of the stack state.

**Comment by marians** [comment]
I got some more feedback in SIG UX and we decided to ask a broader audience.

Here are two versions presented:

image

- (A) would show deleted clusters in the output always, as implemented in this PR. The red highlighting should only occur after at least 24 hours from the delete_date, to differentiate it from a cluster that has been deleted a couple of minutes ago and to indicate "something isn't right".

- (B) Would hide deleted clusters by default, but would show an additional info line at the bottom of the output. Users would have to add `--show-deleted` to list all clusters, including the deleted ones. The result would look like (A).

(B) is currently the winner, so I'll adapt the PR accordingly.

**Comment by marians** [comment]
Regarding sorting, this table is sorted by ID. That has not changed in this PR.

**Comment by marians** [comment]
Adapted the behaviour to what we discussed / what the majority voted for. Please take another look.

**Comment by marians** [comment]
Just found another detail to fix

File: commands/list/clusters/command.go
Comment (pipo02mix) [inline commands/list/clusters/command.go]: I would extract repeated logic before the condition but no strong feelings

**Review by pipo02mix — APPROVED** [review]
LGTM in general, minor comment and also the flag name does not reflect really the purposed (deleted vs deleting) but if none else has concerns, go ahead. Plz wait for sig operator approval

File: commands/list/clusters/command.go
Comment (marians) [reply commands/list/clusters/command.go]: Good point! Changed.

**Comment by marians** [comment]
@pipo02mix If I get you right, you are saying that "deleted" suggests something different than what the user is getting here, correct? Fair point.

**Comment by marians** [comment]
Modified after @pipo02mix 's comment:

image

**Pull request merged**
